### PR TITLE
distributor: include ingester address in push fail error

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

In case of an error it might be hard otherwise to figure out which ingester to investigate.

#### Which issue(s) this PR fixes or relates to

Fixes N/A

#### Checklist

- [X] Tests updated
- N/A Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -1223,9 +1223,12 @@ func (d *Distributor) send(ctx context.Context, ingester ring.InstanceDesc, time
 	_, err = c.Push(ctx, &req)
 	if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
 		// Wrap HTTP gRPC error with more explanatory message.
-		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: %s", resp.Body)
+		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: address=%s body=%s", ingester.Addr, resp.Body)
 	}
-	return errors.Wrap(err, "failed pushing to ingester")
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed pushing to ingester: address=%s", ingester.Addr))
+	}
```
Comment (aknuds1) [inline pkg/distributor/distributor.go:1230]: Is there any point in changing the code to check if `err != nil`, as `errors.Wrap` also handles nil errors? Also, can't you just use `errors.Wrapf` for string formatting?

```suggestion
	return errors.Wrapf(err, "failed pushing to ingester: address=%s", ingester.Addr)
```

Comment (krajorama) [reply pkg/distributor/distributor.go]: you're right, fixed

File: pkg/distributor/distributor.go [hunk 2]
```diff
@@ -1223,9 +1223,9 @@ func (d *Distributor) send(ctx context.Context, ingester ring.InstanceDesc, time
 	_, err = c.Push(ctx, &req)
 	if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
 		// Wrap HTTP gRPC error with more explanatory message.
-		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: %s", resp.Body)
+		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: address=%s body=%s", ingester.Addr, resp.Body)
```
Comment (pracucci) [inline pkg/distributor/distributor.go:1226]: This error will be exposed to the client. We're already exposing internal IPs, but I'm wondering if exposing the body may contain sensitive info (none that comes to my mind right now, to be honest).

Comment (aknuds1) [reply pkg/distributor/distributor.go:1226]: I was also wondering if this might expose the address to the client, and whether it's desirable :)

Comment (krajorama) [reply pkg/distributor/distributor.go:1226]: hmm, true. There's two ways to deal with this: keep the error and log it, but send a sanitized/generic version to the caller. Simplify the error message and log the details directly in the send function. Not sure which one we'd usually do.

Comment (krajorama) [reply pkg/distributor/distributor.go:1226]: Scanning through https://github.com/grafana/mimir/tree/main/docs/internal/contributing there's no guidance on this.

Comment (bboreham) [reply pkg/distributor/distributor.go:1226]: Idea: make sure the info is in a trace span, and make sure the trace ID can be found by the person investigating.

**Comment by krajorama** [comment]
I'll return to this when I get a chance, closing for now.

**Pull request closed without merging**